istioctl/cmd/debug.go

```diff
@@ -0,0 +1,143 @@
+  istioctl x debug syncz --xds-address istio.cloudprovider.example.com:15012
+
+  # Retrieve syncz debug information via Kubernetes config, using token security
+  # (This is the usual way to get the debug information with an in-cluster control plane.)
+  istioctl x debug syncz
+
+  # Retrieve syncz debug information directly from the control plane, using RSA certificate security
+  # (Certificates must be obtained before this step.  The --cert-dir flag lets istioctl bypass the Kubernetes API server.)
+  istioctl x debug syncz --xds-address istio.example.com:15012 --cert-dir ~/.istio-certs
+
+  # Retrieve syncz information via XDS from specific control plane in multi-control plane in-cluster configuration
+  # (Select a specific control plane in an in-cluster canary Istio configuration.)
+  istioctl x debug syncz --xds-label istio.io/rev=default
+`,
+		Aliases: []string{"debug"},
+		RunE: func(c *cobra.Command, args []string) error {
+			kubeClient, err := kubeClientWithRevision(kubeconfig, configContext, opts.Revision)
+			if err != nil {
+				return err
+			}
+			if len(args) == 0 {
+				return CommandParseError{
+					e: fmt.Errorf("debug type is required"),
+				}
+			}
+			if len(args) > 1 {
+				podName, ns, err := handlers.InferPodInfoFromTypedResource(args[1],
+					handlers.HandleNamespace(namespace, defaultNamespace),
+					kubeClient.UtilFactory())
+				if err != nil {
+					return err
+				}
+				pod, err := kubeClient.CoreV1().Pods(ns).Get(context.TODO(), podName, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				serviceAccount = pod.Spec.ServiceAccountName
+				xdsRequest := xdsapi.DiscoveryRequest{
+					ResourceNames: []string{args[0], podName},
+					Node: &envoy_corev3.Node{
+						Id: "debug~0.0.0.0~istioctl~cluster.local",
+					},
+					TypeUrl: pilotxds.TypeDebug,
+				}
+				xdsResponses, _ := multixds.FirstRequestAndProcessXds(&xdsRequest, &centralOpts, istioNamespace,
+					ns, serviceAccount, kubeClient)
+
+				if args[0] == "config_dump" {
+					envoyDump, err := kubeClient.EnvoyDo(context.TODO(), podName, ns, "GET", "config_dump", nil)
+					if err != nil {
+						return fmt.Errorf("could not contact sidecar: %w", err)
+					}
+
+					if err != nil {
+						return err
+					}
+					c, err := compare.NewXdsComparator(c.OutOrStdout(), xdsResponses, envoyDump)
+					if err != nil {
+						return err
+					}
+					return c.Diff()
+				}
+
+			}
+
+			xdsRequest := xdsapi.DiscoveryRequest{
+				ResourceNames: []string{args[0]},
+				Node: &envoy_corev3.Node{
+					Id: "debug~0.0.0.0~istioctl~cluster.local",
+				},
+				TypeUrl: pilotxds.TypeDebug,
+			}
+			xdsResponses, err := multixds.AllRequestAndProcessXds(&xdsRequest, &centralOpts, istioNamespace,
+				istioNamespace, serviceAccount, kubeClient)
+			if err != nil {
+				return err
+			}
+			sw := pilot.XdsStatusWriter{Writer: c.OutOrStdout()}
+			return sw.PrintAll(xdsResponses)
+		},
+	}
+
```

Comment:
This will provide info that this command is experimental. Already added in other experimental commands. See https://github.com/istio/istio/pull/27504
```suggestion
debugCommand.Long += "\n\n" + ExperimentalMsg
```

Reply:
Thanks! will use it